src/content/en/2021/caching.md

```diff
@@ -1,20 +1,488 @@
+
+{{ figure_markup(
+  image="vary-directives.png",
+  caption="Usage of `Vary` directives.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=1279247484&format=interactive",
+  height=436,
+  sheets_gid="1033782866",
+  sql_file="vary_directives.sql"
+) }}
+
+The most popular `Vary` header is `Accept-Encoding` with 90.3% usage, followed by `User-Agent` with 10.9%, `Origin` with 10.1%, and `Accept` with 4.8%.
+
+This shows a 1.5% decrease in use of `Accept-Encoding` from 2020.
+
+{{ figure_markup(
+  caption="Percent of mobile responses that set the `Vary` header.",
+  content="46.3%",
+  classes="big-number",
+  sheets_gid="1033782866",
+  sql_file="vary_directives.sql"
+) }}
+
+It's important to point out that only 46.25% of total requests audited use the `Vary` header, but when compared to 2020, we see an overall increase by 2.85%.
+
+{{ figure_markup(
+  caption="Percent of mobile responses with the `Vary` header that also set `Cache-Control`.",
+  content="83.4%",
+  classes="big-number",
+  sheets_gid="1033782866",
+  sql_file="vary_directives.sql"
+) }}
+
+Of the requests using the `Vary` header, 83.4% also have the `Cache-control`. This shows us a 2.1% improvement from 2020.
+
+## Setting cookies on cacheable responses
+
+In the 2020 Caching chapter, we were reminded to be aware of using `set-cookie` with cacheable responses because only 4.9% of responses used the `private` directive, putting a user's private data at risk of being accidentally served to a different user via a CDN.
+
+{{ figure_markup(
+  image="cacheable-set-cookie.png",
+  caption="Percent of cacheable responses that use `Set-Cookie`.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=261881101&format=interactive",
+  sheets_gid="104392613",
+  sql_file="set_cookie.sql"
+) }}
+
+In 2021, we see an increase in awareness regarding `set-cookie` and caching coexisting. While still only 5% of web pages are using the private directive with `set-cookie`, the total number of cacheable `set-cookie` responses decreased by 4.41%.
+
+## What type of content are we caching?
+
+Font, CSS, and Audio files are over 99% cacheable, with almost 100% of pages currently caching fonts. This could be in part due to their static nature, making them prime choices for caching.
+
+{{ figure_markup(
+  image="caching-by-resource-type.png",
+  caption="The percent of requests that use caching strategies by resource type.",
+  description="TODO",
```

Comment:
description="A bar chart showing distribution of cacheable resource types. In desktop responses, 99.8% of font, 99.3% of CSS, 99.3% of audio, 98.8% of video, 95.3% of scripts, 91.3% of images, 80.2% of xml, 72.6% of html, 65.0% of other types, and 29.8% of text is cacheable. In mobile responses, 99.8% of font, 99.1% of CSS, 99.0% of audio, 98.8% of video, 95.1% of scripts, 89.9% of images, 81.0% of xml, 76.6% of html, 65.4% of other types, and 29.5% of text is cacheable.",